sdk/core/core/src/az_http_request.c

```diff
@@ -179,23 +179,37 @@ az_http_request_get_header(_az_http_request const* request, int32_t index, az_pa
   return AZ_OK;
 }
 
-AZ_NODISCARD az_result az_http_request_get_parts(
-    _az_http_request const* request,
-    az_http_method* out_method,
-    az_span* out_url,
-    az_span* out_body)
+AZ_NODISCARD az_result
+az_http_request_get_method(_az_http_request const* request, az_http_method* out_method)
```

Comment:
Can this just return `az_http_method` rather than `az_result`? When could this ever fail?

Same with the other url and body getters.

So:
```C
AZ_NODISCARD az_http_method az_http_request_get_method(_az_http_request const* request)
```

Reply:
calling this method with one az_http_request that was not init before (having only garbage on it) will return that garbage right now.
If we decide it later, we can validate that the response was actually previously init or return ERROR here.
For that we usually might want to user feedback.
I think current approach let us handle that in the future easier than not returning az_result

Reply:
OK.